README.md

```diff
@@ -2,6 +2,118 @@
 
 Option is a C# representation of the maybe monad
 
+### Monad
```

Comment:
To be honest, I would consider removing this description of a Monad.  There are plenty of descriptions of a Monad all around the web.  In this case, I don't think it helps explain this library.

Reply:
Just been re-thinking this a little.  Perhaps these three sections could be removed and instead we would link to some excellent resources on the web whenever we you mention these terms elsewhere in the README?